### ISSUE TITLE
Error compiling code casting a generic alias

### ISSUE THREAD

**Issue by bitcrowd0** [vlang/v#13864]
**V doctor:**
```
OS: linux, "Fedora release 35 (Thirty Five)"
V full version: V 0.2.4 875ad1f.42a6783
CC version: cc (GCC) 11.2.1 20220127 (Red Hat 11.2.1-9)
```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg report_test.v`
```v
module report

type ParseRes = Result<[]Token, ParseErr>

struct ParseErr{

}

type Opt<T> = None<T> | Some<T>

struct None<T> {}

struct Some<T> {
	value T
}

type Result<T, U> = Err<U> | Ok<T>

struct Ok<T> {
	value T
}

struct Err<U> {
	value U
}

fn test_report() {
	r := Opt<ParseRes>(None<ParseRes>{})
	match r {
		Some<ParseRes> {
			rx := cast(r.value)
		}
		None<ParseRes> {}
	}
}

fn cast(r ParseRes) Result<[]Token, ParseErr> {
	return r
}
```

**What did you expect to see?**

The code should compile

**What did you see instead?**
```
==================
/tmp/v_1000/report_test.11373589862485782346.tmp.c:1539: error: ';' expected (got "Array_report__Token")
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

**Comment by pluspeter0** [comment]
After `Token` is defined, the code compiles without problem.
```v
module report

type ParseRes = Result<[]Token, ParseErr>

struct ParseErr{

}

struct Token{}

type Opt<T> = None<T> | Some<T>

struct None<T> {}

struct Some<T> {
	value T
}

type Result<T, U> = Err<U> | Ok<T>

struct Ok<T> {
	value T
}

struct Err<U> {
	value U
}

fn test_report() {
	r := Opt<ParseRes>(None<ParseRes>{})
	match r {
		Some<ParseRes> {
			rx := cast(r.value)
		}
		None<ParseRes> {}
	}
}

fn cast(r ParseRes) Result<[]Token, ParseErr> {
	return r
}
```

**Comment by Dialga** [comment]
Duplicate of #14002, fixed by #14053.
